src/gen_smtp_client.erl

```diff
@@ -1166,6 +1147,33 @@ session_start_test_() ->
 						end
 					}
 			end,
+			fun({ListenSock}) ->
+					{"Send with callback",
+						fun() ->
+							Options = [{relay, "localhost"}, {port, 9876}, {hostname, <<"testing">>}],
+							Self = self(),
+							Ref = make_ref(),
+							Callback = fun (Arg) -> Self ! {callback, Ref, Arg} end,
+							{ok, _Pid} = send({<<"test@foo.com">>, [<<"foo@bar.com">>], <<"hello world">>}, Options, Callback),
+							{ok, X} = smtp_socket:accept(ListenSock, 1000),
+							smtp_socket:send(X, "220 Some banner\r\n"),
+							?assertMatch({ok, "EHLO testing\r\n"}, smtp_socket:recv(X, 0, 1000)),
+							smtp_socket:send(X, "250 hostname\r\n"),
+							?assertMatch({ok, "MAIL FROM:<test@foo.com>\r\n"}, smtp_socket:recv(X, 0, 1000)),
+							smtp_socket:send(X, "250 ok\r\n"),
+							?assertMatch({ok, "RCPT TO:<foo@bar.com>\r\n"}, smtp_socket:recv(X, 0, 1000)),
+							smtp_socket:send(X, "250 ok\r\n"),
+							?assertMatch({ok, "DATA\r\n"}, smtp_socket:recv(X, 0, 1000)),
+							smtp_socket:send(X, "354 ok\r\n"),
+							?assertMatch({ok, "hello world\r\n"}, smtp_socket:recv(X, 0, 1000)),
+							?assertMatch({ok, ".\r\n"}, smtp_socket:recv(X, 0, 1000)),
+							smtp_socket:send(X, "250 ok\r\n"),
+							?assertMatch({ok, "QUIT\r\n"}, smtp_socket:recv(X, 0, 1000)),
+							?assertMatch({ok, <<"ok\r\n">>}, receive {callback, Ref, CbRet} -> CbRet end),
+							ok
+						end
+					}
```

Comment:
might be good to have a test that tests failure scenario, but I won't insist

Reply:
I'll see to it later.

Reply:
Done :)

Reply:
Thanks 👍